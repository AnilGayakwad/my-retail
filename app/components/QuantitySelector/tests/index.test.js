import React from 'react';
import { shallowWithIntl } from 'utils/intlEnzymeTestHelper';

import { QuantitySelector, mapDispatchToProps } from '../index';

describe('<QuantitySelector />', () => {
  const props = {
    uiState: {
      quantitySelected: 1,
    },
    updateUiState: () => {},
  };

  it('should render QuantitySelector component', () => {
    const renderedComponent = shallowWithIntl(
      <QuantitySelector
        {...props}
      />
    );
    expect(renderedComponent.length).toBe(1);
  });

  describe('updateUiStateObject', () => {
    it('should call updateUiState on updateUiStateObject', () => {
      const mockFunc = jest.fn();
      const newProps = Object.assign({}, props, { updateUiState: mockFunc });
      const renderedComponent = shallowWithIntl(
        <QuantitySelector
          {...newProps}
        />
      );
      renderedComponent.instance().updateUiStateObject(mockFunc);
      expect(mockFunc).toHaveBeenCalled();
    });
  });

  describe('incrementQuantity', () => {
    it('should call updateUiState on incrementQuantity', () => {
      const mockFunc = jest.fn();
      const newProps = Object.assign({}, props, { updateUiState: mockFunc });
      const renderedComponent = shallowWithIntl(
        <QuantitySelector
          {...newProps}
        />
      );
      renderedComponent.instance().incrementQuantity();
      expect(mockFunc).toHaveBeenCalled();
    });
  });

  describe('decrementQuantity', () => {
    it('should call updateUiState on decrementQuantity', () => {
      const mockFunc = jest.fn();
      const newProps = Object.assign({}, props, { updateUiState: mockFunc });
      const renderedComponent = shallowWithIntl(
        <QuantitySelector
          {...newProps}
        />
      );
      renderedComponent.instance().decrementQuantity(mockFunc);
      expect(mockFunc).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps function', () => {
    it('should mapDispatchToProps correctly', () => {
      const dispatchMap = mapDispatchToProps((e) => e);
      expect(dispatchMap).toMatchSnapshot();
    });
  });
});
