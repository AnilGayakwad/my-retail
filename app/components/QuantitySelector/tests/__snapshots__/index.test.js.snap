// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuantitySelector /> mapDispatchToProps function should mapDispatchToProps correctly 1`] = `
Object {
  "updateUiState": [Function],
}
`;
